traitlets/config/configurable.py

```diff
@@ -221,17 +221,20 @@ class defaults.
         base_classes = ','.join(p.__name__ for p in cls.__bases__)
         final_help.append(u'%s(%s) options' % (cls.__name__, base_classes))
         final_help.append(len(final_help[0])*u'-')
-        for k, v in sorted(cls.class_traits(config=True).items()):
+        for k, v in sorted(cls.class_own_traits(config=True).items()):
```

Comment:
No strong opinion on that. 
As reasoned in [this comment](/ipython/traitlets/pull/361#issuecomment-272519698), this is also to avoid clutter on `--help-all`: since #289 class-parents are printed, so the user may connect the dots and deduce which traits belong to subclasses.

But I will retract it, to keep the `--help-all` extra helpfull.